sourcegraph/sourcegraph
# Setup Wizard: Add config alerts and fix redirection

Closes https://github.com/sourcegraph/sourcegraph/issues/46511

This PR adds two things
- Redirect logic for cases when you have no connected code hosts
- Alerts UI for the remote code host setup step that explains different stats of your instance, (1) external configuration is provided, and edits don't work, (2) external configuration is provided, and edits are allowed, but after instance re-start these edits will be lost 

| Configuration provided (edits are disabled) | Configuration provided (edits are allowed) |
| ---------- | ----------- |
| <img width="1135" alt="Screenshot 2023-03-02 at 16 15 59" src="https://user-images.githubusercontent.com/18492575/222538916-a84fd07b-4cec-4bcb-aee2-17efc306e96d.png">| <img width="1135" alt="Screenshot 2023-03-02 at 16 16 19" src="https://user-images.githubusercontent.com/18492575/222539927-d4518f54-46ec-49be-bf21-e26476440c6b.png"> |

## Test plan
- Run instance without any repositories (code hosts) 
- You should see the setup wizard page by default 
- After you finish with setup and click finish, you should see the search page without global banners about setup 
- Run `sg start`. You should see that edits are disabled (blue alert) on the remote code host steps
- Run `EXTSVC_CONFIG_ALLOW_EDITS=1 sg start`, and you should see a yellow alert that edits won't be saved if you re-run your instance
- Run `EXTSVC_CONFIG_FILE='' sg start`. You should see no alerts



## App preview:

- [Web](https://sg-web-vk-fix-setup-wizard-appereance.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
